R/nist.R

```diff
@@ -1,3 +1,128 @@
+#'   multiple matches to a query. Using an identifier other than a CAS number
+#'   will cause this function to run slower as CAS numbers are used as internal
+#'   identifiers by NIST.
+#' @param type Retention index type: \code{"kovats"}, \code{"linear"},
+#'   \code{"alkane"}, and/or \code{"lee"}. See details for more.
+#' @param polarity Column polarity: \code{"polar"} and/or \code{"non-polar"}
+#'   to get RIs calculated for polar or non-polar columns.
+#' @param temp_prog Temperature program: \code{"isothermal"},
+#'   \code{"ramp"}, and/or \code{"custom"}.
+#' @param cas deprecated.  Use \code{query} instead.
+#' @param verbose logical; should a verbose output be printed on the console?
+#' @details The types of retention indices included in NIST include Kovats
+#'   (\code{"kovats"}), Van den Dool and Kratz (\code{"linear"}), normal alkane
+#'   (\code{"alkane"}), and Lee (\code{"lee"}). Details about how these are
+#'   calculated are available on the NIST website:
+#'   \url{https://webbook.nist.gov/chemistry/gc-ri/}
+#' @importFrom purrr map
+#' @importFrom purrr map_dfr
+#' @import dplyr
+#'
+#' @return returns a tibble of literature RIs with the following columns:
+#' \itemize{
+#' \item{\code{query} is the query provided to the NIST server}
+#' \item{\code{cas} is the CAS number or unique record identified used by NIST}
+#' \item{\code{RI} is retention index}
+#' \item{\code{type} is the type of RI (e.g. "kovats", "linear", "alkane", or "lee")}
+#' \item{\code{polarity} is the polarity of the column (either "polar" or "non-polar")}
+#' \item{\code{temp_prog} is the type of temperature program (e.g. "isothermal", "ramp", or "custom")}
+#' \item{\code{column} is the column type, e.g. "capillary"}
+#' \item{\code{phase} is the stationary phase (column phase)}
+#' \item{\code{length} is column length in meters}
+#' \item{\code{gas} is the carrier gas used}
+#' \item{\code{substrate}}
+#' \item{\code{diameter} is the column diameter in mm}
+#' \item{\code{thickness} is the phase thickness in µm}
+#' \item{\code{program}. various columns depending on the value of
+#' \code{temp_prog}}
+#' \item{\code{reference} is where this retention index was published}
+#' \item{\code{comment}. I believe this denotes the database these data
+#'       were aggregated from}
+#'}
+#'
+#' @references NIST Mass Spectrometry Data Center, William E. Wallace, director,
+#'   "Retention Indices" in NIST Chemistry WebBook, NIST Standard Reference
+#'   Database Number 69, Eds. P.J. Linstrom and W.G. Mallard,
+#'   National Institute of Standards and Technology, Gaithersburg MD, 20899,
+#'   \doi{10.18434/T4D303}.
+#'
+#' @export
+#' @note Copyright for NIST Standard Reference Data is governed by the Standard
+#' Reference Data Act, \url{https://www.nist.gov/srd/public-law}.
+#' @seealso \code{\link{is.cas}} \code{\link{as.cas}}
+#'
+#' @examples
+#' \dontrun{
+#' myRIs <- nist_ri(c("78-70-6", "13474-59-4"), from = "cas", "linear",
+#' "non-polar", "ramp")
+#' }
```

Comment:
Let's show off the new feature in the example

```suggestion
#' myRIs <-
#'   nist_ri(
#'     c("78-70-6", "13474-59-4"),
#'     from = "cas",
#'     type = c("linear", "kovats"),
#'     polarity = "non-polar",
#'     temp_prog = "ramp"
#'   )
myRIs
```